packages/grafana-data/src/panel/PanelPlugin.ts

```diff
@@ -203,8 +204,12 @@ export class PanelPlugin<
    *
    * This is a good place to support any changes to the options model
    */
-  setMigrationHandler(handler: PanelMigrationHandler<TOptions>) {
+  setMigrationHandler(
+    handler: PanelMigrationHandler<TOptions>,
+    shouldMigrate?: (panel: ComponentType<PanelProps<TOptions>> | null) => boolean
+  ) {
     this.onPanelMigration = handler;
+    this.shouldMigrate = shouldMigrate;
```

Comment:
can you expand how this `shouldMigrate` is used for panels? how will this affect plugins that are not passing this parameter?

Reply:
it's optional, a way for plugins to have grafana call the onPanelMigration handler even when version is the same

Reply:
Could you add this to the comment in the function? that ways this will show up in editor's signatures when they are calling the method?

Reply:
thanks! I asked because this paramter sets some internal object flag and I am not sure what it does downstream the code.